src/main/java/uk/nhs/hee/tis/common/upload/enumeration/LifecycleState.java

```diff
@@ -0,0 +1,31 @@
+/*
+ * The MIT License (MIT)
+ *
+ * Copyright 2023 Crown Copyright (Health Education England)
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy of this software and
+ * associated documentation files (the "Software"), to deal in the Software without restriction,
+ * including without limitation the rights to use, copy, modify, merge, publish, distribute,
+ * sublicense, and/or sell copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all copies or
+ * substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT
+ * NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+ * NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM,
+ * DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+package uk.nhs.hee.tis.common.upload.enumeration;
+
+/**
+ * An enumeration for object lifecycle state.
+ */
+public enum LifecycleState {
```

Comment:
In the context of "common" object storage, what is "SUBMITTED & UNSUBMITTED"?

It would be good to clarify if we are talking about a field specific to one of the clients of this service.  I'm assuming we're not talking about the "Lifecycle" config that determines S3 storage class.

Reply:
Yeah, it is just a custom indicator to indicate the object is being partially deleted. The `DELETED` state will be set in the object's user metadata. But the TSS also uses this field as trainee form status with `SUBMITTED` and `UNSUBMITTED`.

Only the `DELETED` one is used in the implementation. So I have removed the unused `SUBMITTED` and `UNSUBMITTED` from the enum and added more details in the JavaDoc description.